pkg/scheduler/queue/queue.go

```diff
@@ -128,50 +128,35 @@ func (q *RequestQueue) GetNextRequestForQuerier(ctx context.Context, last UserIn
 	q.mtx.Lock()
 	defer q.mtx.Unlock()
 
-	querierWait := false
-
-FindQueue:
-	// We need to wait if there are no users, or no pending requests for given querier.
-	for (q.queues.len() == 0 || querierWait) && ctx.Err() == nil && !q.stopped {
-		querierWait = false
```

Comment:
how could the number of registered user queues change while we're holding the mutex lock?